subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/CorruptedTaskHistoryIntegrationTest.groovy

```diff
@@ -16,34 +16,69 @@
 
 package org.gradle.api.internal.changedetection
 
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.InputDirectory
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.daemon.DaemonLogsAnalyzer
+import org.gradle.test.fixtures.server.http.BlockingHttpServer
+import org.gradle.workers.WorkerExecutor
+import org.junit.Rule
 import spock.lang.Issue
 
 class CorruptedTaskHistoryIntegrationTest extends AbstractIntegrationSpec {
+    @Rule
+    BlockingHttpServer server = new BlockingHttpServer()
 
     // If this test starts to be flaky it points to a problem in the code!
     // See the linked issue.
     @Issue("https://github.com/gradle/gradle/issues/2827")
     def "broken build doesn't corrupt the artifact history"() {
+        server.start()
         def numberOfFiles = 10
         def numberOfOutputFilesPerTask = numberOfFiles
         def numberOfInputProperties = 10
         def numberOfTasks = 100
         def totalNumberOfOutputDirectories = numberOfTasks
-        def killPollInterval = 10
+        def numberOfExecutedTaskBeforeKilling = 11
         def totalNumberOfOutputFiles = numberOfTasks * numberOfOutputFilesPerTask + totalNumberOfOutputDirectories
 
-        setupTestProject(numberOfFiles, numberOfInputProperties, numberOfTasks, killPollInterval)
+        setupTestProject(numberOfFiles, numberOfInputProperties, numberOfTasks)
 
         executer.beforeExecute {
             // We need a separate JVM in order not to kill the test JVM
             requireGradleDistribution()
+            requireIsolatedDaemons()
+            requireDaemon()
```

Comment:
I think we can get rid of the `requireGradleDistribution()` above as this shouldn't be true. The fixtures can start a daemon without needing a distribution.